### ISSUE TITLE
Chain ID mismatch on testing veYFI

### ISSUE THREAD

**Issue by fubuloubu** [ApeWorX/ape-hardhat#43]
Some wacky error here:

```py
    def test_reward_checkpoint(accounts, ve_yfi_rewards, ve_yfi, panda, gov):
        with ape.reverts("!authorized"):
            ve_yfi_rewards.rewardCheckpoint(panda, sender=panda)
        with ape.reverts("!authorized"):
            ve_yfi_rewards.rewardCheckpoint(panda, sender=gov)
>       ve_yfi_rewards.rewardCheckpoint(panda, sender=ve_yfi)

tests/functional/test_ve_yfi_rewards.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
site-packages/ape_hardhat/providers.py:347: in send_transaction
    txn_hash = self._web3.eth.send_transaction(txn_dict)  # type: ignore
site-packages/web3/eth.py:815: in send_transaction
    return self._send_transaction(transaction)
site-packages/web3/module.py:57: in caller
    result = w3.manager.request_blocking(method_str,
site-packages/web3/manager.py:197: in request_blocking
    response = self._make_request(method, params)
site-packages/web3/manager.py:150: in _make_request
    return request_func(method, params)
site-packages/web3/middleware/formatting.py:94: in middleware
    response = make_request(method, params)
site-packages/web3/middleware/gas_price_strategy.py:89: in middleware
    return make_request(method, (transaction,))
site-packages/web3/middleware/formatting.py:94: in middleware
    response = make_request(method, params)
site-packages/web3/middleware/attrdict.py:33: in middleware
    response = make_request(method, params)
site-packages/web3/middleware/formatting.py:94: in middleware
    response = make_request(method, params)
site-packages/web3/middleware/formatting.py:93: in middleware
    params = formatter(params)
cytoolz/functoolz.pyx:503: in cytoolz.functoolz.Compose.__call__
    ???
cytoolz/functoolz.pyx:250: in cytoolz.functoolz.curry.__call__
    ???
site-packages/eth_utils/decorators.py:91: in wrapper
    return ReturnType(result)  # type: ignore
site-packages/eth_utils/applicators.py:22: in apply_formatter_at_index
    yield formatter(item)
cytoolz/functoolz.pyx:250: in cytoolz.functoolz.curry.__call__
    ???
site-packages/eth_utils/functional.py:45: in inner
    return callback(fn(*args, **kwargs))
site-packages/eth_utils/applicators.py:84: in apply_formatters_to_dict
    yield key, formatters[key](item)
cytoolz/functoolz.pyx:250: in cytoolz.functoolz.curry.__call__
    ???
site-packages/eth_utils/applicators.py:72: in apply_formatter_if
    return formatter(value)
cytoolz/functoolz.pyx:250: in cytoolz.functoolz.curry.__call__
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

web3_chain_id = 31337, chain_id = 0

    @curry
    def _validate_chain_id(web3_chain_id: int, chain_id: int) -> int:
        if to_integer_if_hex(chain_id) == web3_chain_id:
            return chain_id
        else:
>           raise ValidationError(
                "The transaction declared chain ID %r, "
                "but the connected node is on %r" % (
                    chain_id,
                    web3_chain_id,
                )
            )
E           web3.exceptions.ValidationError: The transaction declared chain ID 0, but the connected node is on 31337

site-packages/web3/middleware/validation.py:61: ValidationError
```

_Originally posted by @fubuloubu in https://github.com/ApeWorX/ape/issues/619#issuecomment-1082458528_

**Comment by sabotagebeats** [comment]
I'm getting this error now but wasn't previously

**Comment by unparalleled-js** [comment]
@sabotagebeats  can this be closed now?

**Comment by sabotagebeats** [comment]
@unparalleled-js i believe so.

**Comment by fubuloubu** [comment]
If you can try upgrading that repo to the newest versions of things and confirm nothing is broken before closing, I'd appreciate it!

**Comment by sabotagebeats** [comment]
> If you can try upgrading that repo to the newest versions of things and confirm nothing is broken before closing, I'd appreciate it!

So I have done this with other contracts but not with those specific contracts that you're using. Is this satisfying or should we test with veYFI contracts?

**Comment by sabotagebeats** [comment]
I made a fresh pyenv and cloned veYFI https://github.com/yearn/veYfi
I then ran pip3 install -r requirements.txt`
I then installed solidity vyper and hardhat plugins

all tests from `ape test` in veYFI are passing for me.

The following is my output for versions for ape. 
It may be there needs to be an upgrade of the plugins in veYFI in order to pass the tests. 

```
(veYFI) doge@DESKTOP-E28BJ09:~/ape/veYFI$ ape --version
0.2.4
(veYFI) doge@DESKTOP-E28BJ09:~/ape/veYFI$ ape plugins list
Installed Plugins:
  solidity    0.2.3
  hardhat     0.2.1
  vyper       0.2.0
```
